systemtest/src/test/java/io/strimzi/systemtest/cruisecontrol/CruiseControlST.java

```diff
@@ -806,6 +811,113 @@ void testAutoKafkaRebalanceScaleUpScaleDown() {
+                    .editSpec()
+                        .withStorage(jbodStorage)
+                    .endSpec()
+                    .build(),
+                KafkaNodePoolTemplates.controllerPool(testStorage.getNamespaceName(), testStorage.getControllerPoolName(), testStorage.getClusterName(), 3).build()
+            )
+        );
+
+        // Deploy Kafka with Cruise Control enabled
+        resourceManager.createResourceWithWait(
+            KafkaTemplates.kafkaWithCruiseControlTunedForFastModelGeneration(testStorage.getNamespaceName(), testStorage.getClusterName(), 3, 3)
+                .editOrNewSpec()
+                    .editKafka()
+                        .withStorage(jbodStorage)
+                    .endKafka()
+                .endSpec()
+                .build(),
+            ScraperTemplates.scraperPod(testStorage.getNamespaceName(), testStorage.getScraperName()).build()
+        );
+
+        // Create topics and produce some data to them
+        resourceManager.createResourceWithWait(KafkaTopicTemplates.topic(testStorage.getNamespaceName(), testStorage.getTopicName(), testStorage.getClusterName(), 24, 3).build());
+
+        // send some data to topic
+        final KafkaClients kafkaClients = ClientUtils.getInstantPlainClientBuilder(testStorage).withMessageCount(1000).build();
+        resourceManager.createResourceWithWait(kafkaClients.producerStrimzi(), kafkaClients.consumerStrimzi());
+        ClientUtils.waitForInstantClientSuccess(testStorage);
+
+        final Map<Integer, List<Integer>> brokersWithRemovedVolumes = Map.of(
+            0, Arrays.asList(1, 2),
+            2, Arrays.asList(1)
+        );
+
+        final Map<String, Long> initialDataDirSizes = KafkaTopicUtils.getDataDirectorySizes(testStorage, brokersWithRemovedVolumes);
+        final Map<String, Set<Integer>> initialPartitionDirs = KafkaTopicUtils.getPartitionDirectories(testStorage, brokersWithRemovedVolumes);
+
+        LOGGER.info("Initial data directory sizes: {}", initialDataDirSizes);
+        LOGGER.info("Initial partition directories: {}", initialPartitionDirs);
+
+        // Create a KafkaRebalance resource with 'remove-disks' mode, specifying the brokers and volume IDs
+        resourceManager.createResourceWithoutWait(
+            KafkaRebalanceTemplates.kafkaRebalance(testStorage.getNamespaceName(), testStorage.getClusterName())
+                .editOrNewSpec()
+                    .withMode(KafkaRebalanceMode.REMOVE_DISKS)
+                    .withMoveReplicasOffVolumes(
+                        brokersWithRemovedVolumes.entrySet().stream()
+                            .map(entry -> new BrokerAndVolumeIdsBuilder()
+                                .withBrokerId(entry.getKey())
+                                .withVolumeIds(entry.getValue())
+                                .build()
+                            ).collect(Collectors.toList())
```

Comment:
wouldn't be much clearer to just create list of two `BrokerAndVolumeIdsBuilder`s instead of using Map, collecting it to List etc.? I think it can be a bit confusing when you see it for the first time.

Reply:
Sure that's fair point. Updated